include/property.h

```diff
@@ -0,0 +1,181 @@
+#ifndef __PROPERTY_H__
+#define __PROPERTY_H__
+
+#include <iostream>
+#include <set>
+#include <tuple>
+#include <sstream>
+#include <regex>
+
+namespace clayer {
+
+    // Concepts used for property reading and writing
+    template<typename T>
+    concept bool Streamable = requires(T o, std::ostream& s) {
+        {s << o} -> std::ostream&;
```

Comment:
If this is intended to be used for the logging library in addition to the analyser, can we use a more general definition? Something like {s << o } -> Streamable (e.g. NoRecord does this) It's a bit circular but I think it might capture the idea better.